csharp/src/Google.Protobuf/WellKnownTypes/TimestampPartial.cs

```diff
@@ -147,12 +146,13 @@ public static Timestamp FromDateTime(DateTime dateTime)
         {
             if (dateTime.Kind != DateTimeKind.Utc)
             {
-                throw new ArgumentException("Conversion from DateTime to Timestamp requires the DateTime kind to be Utc", "dateTime");
+                throw new ArgumentException("Conversion from DateTime to Timestamp requires the DateTime kind to be Utc", nameof(dateTime));
             }
             // Do the arithmetic using DateTime.Ticks, which is always non-negative, making things simpler.
-            long secondsSinceBclEpoch = dateTime.Ticks / TimeSpan.TicksPerSecond;
-            int nanoseconds = (int)  (dateTime.Ticks % TimeSpan.TicksPerSecond) * Duration.NanosecondsPerTick;
-            return new Timestamp { Seconds = secondsSinceBclEpoch - BclSecondsAtUnixEpoch, Nanos = nanoseconds };
+            long ticks = dateTime.Ticks;
+            long secondsSinceBclEpoch = ticks / TimeSpan.TicksPerSecond;
+            ticks -= secondsSinceBclEpoch * TimeSpan.TicksPerSecond;
```

Comment:
I think this would be clearer as:

```csharp
long subsecondTicks = ticks - (secondsSinceBclEpoch * TimeSpan.TicksPerSecond);
```

(Is this definitely faster than `%`? Would Math.DivRem be better? It seems custom-made for this...)

Reply:
You're right, it would be better to use DivRem.

`long secondsSinceBclEpoch = Math.DivRem(dateTime.Ticks, TimeSpan.TicksPerSecond, out long remainTicks);
return new Timestamp { Seconds = secondsSinceBclEpoch - BclSecondsAtUnixEpoch, Nanos = (int)remainTicks * Duration.NanosecondsPerTick };`

Reply:
Yup. I'd change `remainTicks` to `tickOfSecond` or `subsecondTicks`, but other than that, that sounds good to me.

Reply:
Please could you rebase the branch to HEAD, and then make this change? It would be good to get this in.

Reply:
Looks like this change is still missing. Please could you update the PR?